docs/src/starknet/script.md

```diff
@@ -0,0 +1,176 @@
+# Cairo Deployment Scripts
+
+## Overview
+
+> ⚠️⚠️⚠️ Highly experimental code, a subject to change  ⚠️⚠️⚠️
+
+Starknet Foundry cast can be used to run deployment scripts written in Cairo, using `script` subcommand. 
+It aims to provide similar functionality to Foundry's `forge script`. 
+
+To start writing a deployment script in Cairo just add `cast_std` as a dependency to you scarb package and make sure to
+have a `main` function in the module you want to run.
+
+Please note that **`sncast script` is in develoment**. While it is already possible to declare, deploy, invoke and call 
+contracts from within Cairo, its interface, internals and feature set can change rapidly each version. 
+
+Some of the planned features that will be included in future versions are:
+
+- scripts idempotency
+- dispatchers support
+- better error handling
+- logging
+- account creation/deployment
+- multicall support
+- dry running the scripts
+- init subcommand
+
+and more!
+
+## Examples
+
+### Minimal example (without contract deployment)
+
+This example shows how to call an already deployed contract. Please find full example with contract deployment [here](#full-example-with-contract-deployment).
+
+```cairo
+use sncast_std::{invoke, call, InvokeResult, CallResult};
+
+fn main() {
+    let eth = 0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7;
+    let addr = 0x0089496091c660345BaA480dF76c1A900e57cf34759A899eFd1EADb362b20DB5;
+    let call_result = call(eth.try_into().unwrap(), 'allowance', array![addr, addr]);
+    let call_result = *call_result.data[0];
+    assert(call_result == 0, call_result);
+
+    let call_result = call(eth.try_into().unwrap(), 'decimals', array![]);
+    let call_result = *call_result.data[0];
+    assert(call_result == 18, call_result);
+}
+```
+
+The script should be included in a scarb package. The directory structure and config for this example looks like this:
+
+```shell
+$ tree
+.
+├── src
+│   ├── my_script.cairo
+│   └── lib.cairo
+└── Scarb.toml
+```
+
+```toml
+[package]
+name = "my_script"
+version = "0.1.0"
+
+[dependencies]
+starknet = ">=2.3.0"
+sncast_std = { git = "https://github.com/foundry-rs/starknet-foundry.git", tag = "v0.11.0" }
+```
+
+To run the script, do:
+
+```shell
+$ sncast \
+  --rpc-url http://127.0.0.1:5050 \
```

Comment:
```suggestion
  --url http://127.0.0.1:5050 \
```
https://foundry-rs.github.io/starknet-foundry/appendix/cast/common.html#--url--u-rpc_url

Reply:
great catch! chnged in newest commit